pkg/util/proto/document.go

```diff
@@ -217,30 +243,40 @@ func (d *Definitions) parseKind(s *openapi_v2.Schema, path *Path) (Schema, error
 	}, nil
 }
 
-func (d *Definitions) parseArbitrary(s *openapi_v2.Schema, path *Path) (Schema, error) {
+func (d *definitions) parseArbitrary(s *openapi_v2.Schema, path *Path) (Schema, error) {
 	return &Arbitrary{
 		BaseSchema: d.parseBaseSchema(s, path),
 	}, nil
 }
 
 // ParseSchema creates a walkable Schema from an openapi schema. While
 // this function is public, it doesn't leak through the interface.
-func (d *Definitions) ParseSchema(s *openapi_v2.Schema, path *Path) (Schema, error) {
+func (d *definitions) ParseSchema(s *openapi_v2.Schema, path *Path) (Schema, error) {
 	if s.GetXRef() != "" {
+		// TODO(incomplete): ignoring the rest of s is wrong. As long as there are no conflict, everything from s must be considered
+		// Reference: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#path-item-object
 		return d.parseReference(s, path)
 	}
 	objectTypes := s.GetType().GetValue()
 	switch len(objectTypes) {
 	case 0:
-		// in the OpenAPI schema served by older k8s versions, object definitions created from structs did not include
-		// the type:object property (they only included the "properties" property), so we need to handle this case
-		if s.GetProperties() != nil {
-			return d.parseKind(s, path)
-		} else {
-			// Definition has no type and no properties. Treat it as an arbitrary value
-			// TODO: what if it has additionalProperties or patternProperties?
-			return d.parseArbitrary(s, path)
+		// at the root of schemas we forgot the type somewhen between 1.11 and 1.13, to be fixed again for 1.14.
```

Comment:
so the `if s.GetProperties() != nil` check *only* applied to root-level structs? Is there a test of a schema from "in the OpenAPI schema served by older k8s versions" that exercises this path?